[CI] IndexRecoveryIT testUsesFileBasedRecoveryIfOperationsBasedRecoveryWouldBeUnreasonable failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10907 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/5dekxjk6bq24g)
- [elasticsearch-pull-request #95948 / part-1](https://gradle-enterprise.elastic.co/s/cky53yzaqlz76)
- [elasticsearch-pull-request #95870 / part-1](https://gradle-enterprise.elastic.co/s/7abt5zq5i5pig)
- [elasticsearch-pull-request #95766 / part-1](https://gradle-enterprise.elastic.co/s/ozxmi7yfaz3e4)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.indices.recovery.IndexRecoveryIT.testUsesFileBasedRecoveryIfOperationsBasedRecoveryWouldBeUnreasonable" -Dtests.seed=5A7C2DD8A952A4DA -Dtests.locale=en-DG -Dtests.timezone=Pacific/Palau -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.indices.recovery.IndexRecoveryIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testUsesFileBasedRecoveryIfOperationsBasedRecoveryWouldBeUnreasonable'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: timed out waiting for green state
```

**Issue Reasons:**
- [main] 4 failures in test testUsesFileBasedRecoveryIfOperationsBasedRecoveryWouldBeUnreasonable (1.6% fail rate in 255 executions)
- [main] 3 failures in step part-1 (2.2% fail rate in 134 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (2.3% fail rate in 133 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -414,9 +414,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT
   method: testAggTooManyMvLongs
   issue: https://github.com/elastic/elasticsearch/issues/135585
-- class: org.elasticsearch.indices.recovery.IndexRecoveryIT
-  method: testUsesFileBasedRecoveryIfOperationsBasedRecoveryWouldBeUnreasonable
-  issue: https://github.com/elastic/elasticsearch/issues/135737
 - class: org.elasticsearch.xpack.esql.action.CrossClusterQueryWithPartialResultsIT
   method: testOneRemoteClusterPartial
   issue: https://github.com/elastic/elasticsearch/issues/124055

